src/managers/contact_manager.py

```diff
@@ -48,7 +48,12 @@ def remove_contact(self, name: str) -> None:
         """
         Removes a contact from the list by name.
         """
-        raise NotImplementedError("The 'remove_contact' method is not implemented.")
+        for contact in self.contacts: # Sorting through the contacts in the list
+            if contact.name == name: # Compare contacts
+                self.contacts.remove(contact) # The found contact is deleted
+                return
```

Comment:
@viktoria-nika hello!
Before the “return” you need to add an informative message, something like: f “Contact ‘{name}’ successfully deleted.”
And if nothing is found, then: f “Contact ‘{name}’ not found.”

But I also recommend looking at the generator in the direction of optimization, which will simplify the code, make it clearer, and optimize the execution time.
What I mean is an example:
contact_to_remove = next((contact for contact in self.contacts if contact.name == name), None)

If contact_to_remove finds something, we remove it and display a success message, otherwise, we fail.